File: app/src/main/java/com/foreverrafs/radiocore/data/Serializers.java
package com.foreverrafs.radiocore.data;

import android.util.Log;

import com.foreverrafs.radiocore.model.News;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import org.joda.time.DateTime;

import java.lang.reflect.Type;

public class Serializers {

    static class NewsDeserializer implements JsonDeserializer<News> {
        @Override
        public News deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
            String headline = ((JsonObject) json).get("headline").getAsString();
            String content = ((JsonObject) json).get("content").getAsString();
            String category = ((JsonObject) json).get("category").getAsString();
            String imageUrl = ((JsonObject) json).get("imageUrl").getAsString();
            String dateStr = ((JsonObject) json).get("date").getAsString();
            DateTime date = DateTime.now();

            try {
                date = DateTime.parse(dateStr);
            } catch (Exception exception) {
                Log.i("Serializers", exception.getMessage());
            }

            return new News(headline, date, imageUrl, content, category);
        }
    }

    static class NewsSerializer implements JsonSerializer<News> {

        @Override
        public JsonElement serialize(News src, Type typeOfSrc, JsonSerializationContext context) {
            return null;
        }
    }
}

File: news/src/main/java/com/radiocore/news/util/Serializers.kt
package com.radiocore.news.util

import com.google.gson.*
import com.radiocore.news.data.entities.NewsEntity
import org.joda.time.DateTime
import timber.log.Timber
import java.lang.reflect.Type

class Serializers {
    class NewsDeserializer : JsonDeserializer<NewsEntity> {
        @Throws(JsonParseException::class)
        override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): NewsEntity {
            val headline = (json as JsonObject)["headline"].asString
            val content = json["content"].asString
            val category = json["category"].asString
            val imageUrl = json["imageUrl"].asString
            val dateStr = json["date"].asString
            var date = DateTime.now()
            try {
                date = DateTime.parse(dateStr)
            } catch (exception: Exception) {
                Timber.i(exception)
            }
            return NewsEntity(headline, content, date!!, category, imageUrl)
        }
    }

    class NewsSerializer : JsonSerializer<NewsEntity?> {
        override fun serialize(src: NewsEntity?, typeOfSrc: Type, context: JsonSerializationContext): JsonElement? {
            return null
        }
    }
}